src/main/java/com/uid2/admin/Main.java

```diff
@@ -311,12 +309,17 @@ public void run() {
             DataStoreMetrics.addDataStoreMetrics("salt", saltProvider);
             DataStoreMetrics.addDataStoreMetrics("partners", partnerConfigProvider);
 
+            ReplaceSharingTypesWithSitesJob replaceSharingTypesWithSitesJob = new ReplaceSharingTypesWithSitesJob(config, writeLock, adminKeysetProvider, keysetProvider, keysetStoreWriter, siteProvider);
+            jobDispatcher.enqueue(replaceSharingTypesWithSitesJob);
+            CompletableFuture<Boolean> replaceSharingTypesWithSitesJobFuture = jobDispatcher.executeNextJob();
+            replaceSharingTypesWithSitesJobFuture.get();
+
             //UID2-575 set up a job dispatcher that will write private site data periodically if there is any changes
             //check job for every minute
-            PrivateSiteDataSyncJob job = new PrivateSiteDataSyncJob(config, writeLock);
-            jobDispatcher.enqueue(job);
-            jobDispatcher.executeNextJob();
-
+            PrivateSiteDataSyncJob privateSiteDataSyncJob = new PrivateSiteDataSyncJob(config, writeLock);
+            jobDispatcher.enqueue(privateSiteDataSyncJob);
+            CompletableFuture<Boolean> privateSiteDataSyncJobFuture = jobDispatcher.executeNextJob();
+            privateSiteDataSyncJobFuture.get();
```

Comment:
Previously it did not wait but now it will?

Reply:
Yep, this was basically the reason why we needed to wait 1min after starting admin/core before starting operators locally

If we call `executeNextJob()` back-to-back immediately, the second call won't work because the first job is still running. So you would have to wait 1min for the next job to be executed